pkg/autodiscover/autodiscover_podset.go

```diff
@@ -94,31 +96,33 @@ func findStatefulSetByLabel(
 	labels []labelObject,
 	namespaces []string,
 ) []appsv1.StatefulSet {
-	statefulsets := []appsv1.StatefulSet{}
+	allStatefulSets := []appsv1.StatefulSet{}
 	for _, ns := range namespaces {
-		ss, err := appClient.StatefulSets(ns).List(context.TODO(), metav1.ListOptions{})
+		statefulSet, err := appClient.StatefulSets(ns).List(context.TODO(), metav1.ListOptions{})
 		if err != nil {
 			log.Error("Failed to list statefulsets in ns=%s, err: %v . Trying to proceed.", ns, err)
 			continue
 		}
-		if len(ss.Items) == 0 {
+		if len(statefulSet.Items) == 0 {
 			log.Warn("Did not find any statefulSet in ns=%s", ns)
 		}
-
-		for i := 0; i < len(ss.Items); i++ {
+		for i := 0; i < len(statefulSet.Items); i++ {
```

Comment:
To follow the same pattern as for pods and operator, create a findStatefulSetsMatchingAtLeastOneLabel function

Reply:
Finally, the 'isStatefulSetsMatchingAtLeastOneLabel' function has been created, returning a boolean value.